linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_warn_age_existing/rule.yml

```diff
@@ -0,0 +1,43 @@
+documentation_complete: true
+
+prodtype: sle12,sle15
+
+title: 'Set Existing Passwords Warning Age'
+
+description: |-
+    Configure non-compliant accounts, and to give users time of think of a secure password,
+    to receive 7 days or more an advance warning that a password will be expiring by running 
+    the following command:
+    <pre>$ sudo chage --warndays 7<i>USER</i></pre>
+
+
+rationale: |-
+    Providing an advance warning that a password will be expiring gives users 
+    time to think of a secure password. Users caught unaware may choose a simple 
+    password or write it down where it may be discovered.
+
+severity: medium
+
+identifiers:
+    cce@sle12: CCE-92321-9
+    cce@sle15: CCE-92479-5
+
+references:
+    cis@sle12: 5.4.1.4
+    cis@sle15: 5.4.1.4
+    disa: CCI-000198
+    nist: IA-5(f),IA-5(1)(d),CM-6(a)
+    nist@sle15: IA-5(1).1(v)
+
+
+ocil_clause: 'any results are returned that are not associated with a system account'
+
+ocil: |-
+    Verify that {{{ full_name }}} has configured the warning that a password will be expiring for each user account is seven days or greater with the following command:
+
+    $ sudo awk -F: '$6 &lt; 7 {print $1 " " $6}' /etc/shadow
+
+fixtext: |-
+    Configure non-compliant accounts to enforce a 24 hours/1 day minimum password lifetime:
+
+    $ sudo chage --warndays {{{ xccdf_value("var_accounts_password_warn_age_login_defs") }}} [user]
```

Comment:
Maybe the command here could be more automated, like in the remediation, which detects the users and fix what is necessary in a loop.